internal/clients/interface.go

```diff
@@ -79,6 +80,9 @@ type EC2 interface {
 	RunInstances(ctx context.Context, name *string, amount int32, instanceType types.InstanceType, AMI string, keyName string, userData []byte) ([]*string, *string, error)
 
 	GetAccountId(ctx context.Context) (string, error)
+
+	// ListAttachedRolePolicies returns list of all policies for the given arn role
+	ListAttachedRolePolicies(ctx context.Context, auth *Authentication) ([]*iamTypes.AttachedPolicy, error)
 }
```

Comment:
We must not expose internal types through our EC2 interface, it is here to hide implementation details. Similarly as in other fucntions, please create a new type in this `clients` package and copy relevant values you want to carry over to the caller. Typically it is just minimum set of fields you need to do what you want to achieve. Perhaps some sort of "id" in this case.

But before you do that, consider moving the logic from service package in here. In that case a simple function would do the check and return some struct with result and error (for runtime errors):

```go
struct PolicyCheckResult struct {
  Valid bool
  Missing []string
}

func CheckPolicy(ctx) (PolicyCheckResult, error)
```

That is a different approach, not that flexible, however, it will save you time creating all those custom types. Whatever you like.

Reply:
I like the idea of returning an array of missing permissions. We should not expose anything we do not want. Let me know if the current approach is what we want.